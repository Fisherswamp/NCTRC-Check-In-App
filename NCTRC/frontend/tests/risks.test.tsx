
import React from "react";
import  RiskScreen from '../screens/RisksScreen';
import { shallow, configure } from 'enzyme'; 
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const props = {
    navigation: {
        addListener: jest.fn(), 
        canGoBack: jest.fn(), 
        dangerouslyGetParent: jest.fn(), 
        dangerouslyGetState: jest.fn(), 
        dispatch: jest.fn(), 
        goBack: jest.fn(), 
        isFocused: jest.fn(), 
        navigate: jest.fn(), 
        pop: jest.fn(), 
        popToTop: jest.fn(), 
        push: jest.fn(), 
        removeListener: jest.fn(), 
        replace: jest.fn(), 
        reset: jest.fn(), 
        setOptions: jest.fn(), 
        setParams: jest.fn(), 
    }, 
    route: {
        params: {
            firstName: "string",
            lastName: "string",
            email: "string",
            tempurature: 0,
        }
    }, 
}

describe('Risk Screen renders', () => {
    it('Risk Screen  render', () => {
      const tree = renderer.create(<RiskScreen navigation={props.navigation} route={props.route} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});


describe('Risk elements', () => {
    it("Risk Screen elements", () => {
       let wrapper =  shallow(<RiskScreen navigation={props.navigation} route={props.route} />);
       expect(wrapper.find('Image').length).toBe(2); 
       expect(wrapper.find('Text').length).toBe(10); 
       expect(wrapper.find('View').length).toBe(2); 
    })
   
});

jest.useFakeTimers(); 