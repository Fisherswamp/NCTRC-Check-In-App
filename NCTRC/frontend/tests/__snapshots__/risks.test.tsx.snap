// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homescreen renders App render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/NCTRClogo.png",
        }
      }
      style={
        Object {
          "height": 300,
          "width": 300,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 20,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Check In
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#884633",
          "borderRadius": 9,
          "height": 60,
          "margin": 20,
          "opacity": 1,
          "width": 230,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginTop": 18,
          }
        }
      >
        Check Out
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 150,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#884633",
              },
            ]
          }
        >
          Sign in as Admin
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Risk Screen renders Risk Screen  render 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "marginTop": 40,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "flex-start",
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/backbutton.png",
          }
        }
        style={
          Object {
            "height": 75,
            "width": 75,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "marginTop": -40,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/NCTRClogo.png",
          }
        }
        style={
          Object {
            "height": 150,
            "width": 150,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "bold",
            "marginLeft": 25,
            "marginRight": 25,
            "padding": 20,
          }
        }
      >
        COVID-19 Acknowledgement of Risk and Acceptance of Services
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "marginLeft": 25,
            "marginRight": 25,
            "padding": 20,
          }
        }
      >
        I, 
        string
         
        string
        , am aware of the risks of contracting or spreading Covid-19 while working or volunteering at NC Therapeutic Riding Center; attending an event; and/or receiving face-to-face services from NC Therapeutic Riding Center during the time of a pandemic outbreak, and /or North Carolina Governor’s or Orange County’s declaration of a “stay-at-home” order(s). 
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "marginLeft": 25,
            "marginRight": 25,
            "padding": 20,
          }
        }
      >
        I am aware that face-to-face services and experiences increase my risk of contracting and passing on the Covid-19 or Coronavirus and agree to hold harmless NC Therapeutic Riding Center and its residents, members, officers, managers, agents, employees and all other individuals I may come in contact with during this interaction and receiving of services, providing services, attending an event or volunteering within this organization. I am aware of the options that may be available for remote services including, telephonic and video telehealth, as allowed by insurances and State Licensing Board recommendations, during this Pandemic outbreak.
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "marginLeft": 25,
            "marginRight": 25,
            "padding": 20,
          }
        }
      >
        I agree to and will follow all guidelines for personal hygiene, personal safety and public safety as recommended by NC Therapeutic Riding Center; as well as my individual provider/practitioner. This may include, but is not limited to, waiting in my vehicle and/or home until I am asked to enter the building/farm; maintaining social distance; washing my hands prior to and following each session or activity; use of hand sanitizer upon request; wiping down surfaces with disinfecting wipes and/or wearing a protective medical mask and/or gloves.
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "marginLeft": 25,
            "marginRight": 25,
            "padding": 20,
          }
        }
      >
        I agree to stay home and/or cancel my services should I have personally exhibited or have been in contact with someone who has presented with illness within the previous 24 hours to 2 weeks, including; cough, sneezing, fever, chest congestion or additional signs of potential spread of any virus or bacteria/disease. In addition, I will follow the recommendations of my provider once I have notified them of these risks in regards to my future services or attendance during this pandemic.
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "marginLeft": 25,
            "marginRight": 25,
            "padding": 20,
          }
        }
      >
        NC Therapeutic Riding Center will engage in regular cleaning and sanitizing of the facility, horse tack, grooming supplies and office, doors, and frequently touched areas in-between clients and on a daily basis as recommended by the CDC for the safety of clients, employees, volunteers and horses.
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "marginLeft": 25,
            "marginRight": 25,
            "padding": 20,
          }
        }
      >
        I am signing under my own free will and agree to follow these and hold harmless all individuals associated with or through my services acquired from NC Therapeutic Riding Center.
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "marginLeft": 25,
            "marginRight": 25,
            "padding": 20,
          }
        }
      >
        BY SIGNING BELOW, I CONFIRM THAT I HAVE READ AND UNDERSTAND THIS DOCUMENT.
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "marginLeft": 25,
            "marginRight": 25,
            "padding": 20,
          }
        }
      >
        *In the event that the undersigned is under the age of 18, the signature of a parent or guardian is required.
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#884633",
            "borderRadius": 9,
            "height": 60,
            "margin": 20,
            "marginBottom": 50,
            "opacity": 1,
            "width": 180,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "marginTop": 18,
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
